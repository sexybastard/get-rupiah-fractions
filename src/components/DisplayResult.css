.result-container {
  padding-left: 20px;
  padding-right: 20px;
  margin-top: 17px;
}

.result-container label {
  font-size: 18;
  font-weight: bold;
}

.font-result {
  color: #333;
  font-style: italic;
}

.font-error {
  color: red;
}