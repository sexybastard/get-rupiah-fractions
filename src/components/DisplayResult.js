import React from 'react';
import './DisplayResult.css'

const DisplayResult = ({ result, validateResponse }) => {
  const className = validateResponse.isValid ? 'font-result' : 'font-error'
  const displayText = validateResponse.isValid ? result : validateResponse.message

  return (
    <div id="result-container" className="result-container">
      <label className={className}>{displayText}</label>
    </div>
  )
}

export default DisplayResult;
