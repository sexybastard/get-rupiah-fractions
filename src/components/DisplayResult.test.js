import React from 'react';
import { shallow } from 'enzyme';
import DisplayResult from './DisplayResult';

describe("Component: DisplayResult should match snapshoot", () => {
  it("Case 1", () => {
    const validateResponse = {
      isValid: false,
      message: 'Invalid format'
    }
    const wrapper = shallow(<DisplayResult validateResponse={validateResponse} />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it("Case 2", () => {
    const result = '5x Rp100000, 1xRp50000, left 12'
    const validateResponse = {
      isValid: true,
      message: ''
    }
    const wrapper = shallow(<DisplayResult result={result} validateResponse={validateResponse} />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
})

describe("Component: DisplayResult should have the correct className and content", () => {
  it('Ensure #result-container to have className result-container', () => {
    let validateResponse = {
      isValid: false
    }
    let wrapper = shallow(<DisplayResult validateResponse={validateResponse} />);
    expect(wrapper.find('#result-container').prop('className')).toEqual('result-container')
  });

  it('Ensure #result-container label to have a correct className', () => {
    let validateResponse = {
      isValid: false
    }
    let wrapper = shallow(<DisplayResult validateResponse={validateResponse} />);
    expect(wrapper.find('#result-container label').prop('className')).toEqual('font-error')

    validateResponse = {
      isValid: true
    }
    wrapper = shallow(<DisplayResult validateResponse={validateResponse} />);
    expect(wrapper.find('#result-container label').prop('className')).toEqual('font-result')
  });

  it('Ensure #result-container label to have a correct text', () => {
    let result = '5x Rp100000, 1xRp50000, left 12'
    let validateResponse = {
      isValid: false,
      message: 'Invalid format'
    }
    let wrapper = shallow(<DisplayResult validateResponse={validateResponse} />);
    expect(wrapper.find('#result-container label').text()).toEqual(validateResponse.message)

    validateResponse = {
      isValid: true,
      message: ''
    }
    wrapper = shallow(<DisplayResult result={result} validateResponse={validateResponse} />);
    expect(wrapper.find('#result-container label').text()).toEqual(result)
  });

});
