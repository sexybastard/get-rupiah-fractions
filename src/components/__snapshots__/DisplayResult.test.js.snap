// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: DisplayResult should match snapshoot Case 1 1`] = `
Array [
  <div
    className="result-container"
    id="result-container"
  >
    <label
      className="font-error"
    >
      Invalid format
    </label>
  </div>,
]
`;

exports[`Component: DisplayResult should match snapshoot Case 2 1`] = `
Array [
  <div
    className="result-container"
    id="result-container"
  >
    <label
      className="font-error"
    >
      Invalid format
    </label>
  </div>,
]
`;
